datadog_checks_base/tests/base/utils/db/test_util.py

```diff
@@ -53,12 +53,23 @@ def test_constant_rate_limiter():
     start = time.time()
     sleep_count = 0
     while time.time() - start < test_duration_s:
-        ratelimiter.sleep()
+        ratelimiter.update_last_time_and_sleep()
         sleep_count += 1
     max_expected_count = rate_limit * test_duration_s
     assert max_expected_count - 1 <= sleep_count <= max_expected_count + 1
 
 
+def test_constant_rate_limiter_shell_execute():
```

Comment:
it's good to test the rate limiter change itslef. it will be better to also have a test to cover rate limit in sync job

Reply:
yes, there is a bit of a technical depth in the testing of DBMAsyncJob. As I had many setbacks with kube/aws schema set up I didnt advance as much as I wanted. So, I tried to do a bare minimum i.e - fix existing tests. Let me first add more tests for schema PR.  I'd rather create a follow up ticket, unless I finish schemas till the end of the week.